### DISCUSSION TITLE
running reth in archive mode

### DISCUSSION THREAD

**Discussion by abhijeetbhagat** [paradigmxyz/reth#3764] [Help]
i want to run reth in archive mode so that it only does indexing. the reth book doesnt seem to have anything related.

**Comment by merklefruit** [comment]
Reth is an Ethereum execution client, not (only) an indexer. Although given its composability, it could be turned into one with minimal changes, and there are several ongoing experiments to do so. Some examples and implementation of this concept:

- [DB access example](https://github.com/paradigmxyz/reth/blob/main/examples/db-access.rs)
- [Reth indexer to Postgres DB](https://github.com/joshstevens19/reth-indexer)
- [Reth DB access in python](https://github.com/gibz104/reth-db-py)
- [Reth DB middleware for Ethers-rs](https://github.com/SorellaLabs/ethers-reth)

---

If instead, by "archive mode" you meant keeping all of Ethereum's history inside its DB, then you should know that this is already supported, and full node sync will be added in the future.
